src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -158,3 +164,28 @@ def _resolve_multi(interpreter, requirements, platforms, find_links):
       allow_prereleases=python_setup.resolver_allow_prereleases)
 
   return distributions
+
+
+def build_req_libs_provided_by_setup_file(context, local_built_dists, class_name):
+  """Build a requirements library from a local wheel.
+
+  :param context: The context of the calling task needed for injecting synthetic targets.
+  :param local_built_dists: A list of paths to locally built wheels to package into
+  requirements libraries.
+  :param class_name: The name of the calling task class for naming synthetic targets.
+  :return: a :class: `PythonRequirementLibrary` containing a local wheel and its
+  transitive dependencies.
+  """
+  req_libs = []
+  local_whl_reqs = []
+  for whl_location in local_built_dists:
+    base = os.path.basename(whl_location)
+    whl_dir = os.path.dirname(whl_location)
+    whl_metadata = base.split('-')
+    req_name = '=='.join([whl_metadata[0], whl_metadata[1]])
+    local_whl_reqs.append(PythonRequirement(req_name, repository=whl_dir))
+  if local_whl_reqs:
+    addr = Address.parse(class_name)
```

Comment:
the class name feels like an odd `Address` to use here? it won't be unique, for one - so this likely needs to be generative into some synthetic namespace.

Reply:
✅